### ISSUE TITLE
`InstanceNorm1d` does not match documentation and is redundant.

### ISSUE THREAD

**Issue by veugene** [pytorch/pytorch#9776]
`InstanceNorm1d` expects only 3D inputs, while the documentation states:

"Applies Instance Normalization over a 2D or 3D input (a mini-batch of 1D inputs with optional additional channel dimension)"

Furthermore, `InstanceNorm1d`, `InstanceNorm2d`, and `InstanceNorm3d` appear to be redundant as they add nothing to their parent class `_InstanceNorm` except an input dimension check.

Proposal A:
Fix the dimension check in `InstanceNorm1d` to allow both 2D and 3D inputs.

Proposal B:
Merge all instance norm classes into a single class, for any input dimensions > 1.

**Comment by veugene** [comment]
`BatchNorm1d`, `BatchNorm2d`, and `BatchNorm3d` are similarly redundant.

**Comment by arogozhnikov** [comment]
(confirm that in 0.4 InstanceNorm1d doesn't work with 2D inputs)

**Comment by soumith** [comment]
let's go with proposal A. We planned to deprecate (not delete) `InstanceNorm*d` in favor of a universal `InstanceNorm`, but haven't gotten around to it.
